tests/cmds/test_devices.py

```diff
@@ -0,0 +1,478 @@
+from datetime import date
+
+import pytest
+from pandas import DataFrame
+from pandas import testing
+from py42.exceptions import Py42BadRequestError
+from py42.exceptions import Py42ForbiddenError
+from py42.exceptions import Py42NotFoundError
+from py42.response import Py42Response
+from requests import HTTPError
+from requests import Response
+
+from code42cli import PRODUCT_NAME
+from code42cli.cmds.devices import _add_usernames_to_device_dataframe
+from code42cli.cmds.devices import (
+    _drop_devices_which_have_not_connected_in_some_number_of_days,
+)
+from code42cli.cmds.devices import _drop_n_devices_per_user
+from code42cli.cmds.devices import _get_device_dataframe
+from code42cli.main import cli
+
+_NAMESPACE = "{}.cmds.devices".format(PRODUCT_NAME)
+TEST_DEVICE_ID = "12345"
+TEST_ARCHIVE_GUID = "954143426849296547"
+TEST_PURGE_DATE = "2020-10-12"
+TEST_ARCHIVES_RESPONSE = [
+    {
+        "archives": [
+            {
+                "archiveGuid": "954143426849296547",
+                "userId": None,
+                "userUid": None,
+                "archiveBytes": 1745757673,
+                "targetGuid": "632540230984925185",
+                "lastCompletedBackup": "2020-10-12T20:17:52.084Z",
+                "isColdStorage": False,
+                "lastMaintained": "2020-10-10T19:31:05.811Z",
+                "maintenanceDuration": 455,
+                "compactBytesRemoved": 0,
+                "storePointId": 1000,
+                "selectedBytes": 1658317953,
+                "selectedFiles": 596,
+                "todoBytes": 0,
+                "format": "ARCHIVE_V1",
+            }
+        ]
+    }
+]
+TEST_DEVICE_RESPONSE = """{"data":{"computerId":139527,"name":"testname","osHostname":"testhostname","guid":"954143368874689941","type":"COMPUTER","status":"Active","active":true,"blocked":false,"alertState":0,"alertStates":["OK"],"userId":203988,"userUid":"938960273869958201","orgId":3099,"orgUid":"915323705751579872","computerExtRef":null,"notes":null,"parentComputerId":null,"parentComputerGuid":null,"lastConnected":"2020-10-12T16:55:40.632Z","osName":"win","osVersion":"10.0.18362","osArch":"amd64","address":"172.16.208.140:4242","remoteAddress":"72.50.201.186","javaVersion":"11.0.4","modelInfo":null,"timeZone":"America/Chicago","version":1525200006822,"productVersion":"8.2.2","buildVersion":26,"creationDate":"2020-05-14T13:03:20.302Z","modificationDate":"2020-10-12T16:55:40.632Z","loginDate":"2020-10-12T12:54:45.132Z","service":"CrashPlan"}}"""
```

Comment:
Like a similar comment elsewhere, if you manually make this super long lines multiline, our formatter make it look the way it wants and still keep it multiline... Some of these are quite long!